### PR TITLE
Add node-to-node SSL/HBA integration

### PR DESCRIPTION
**Pull request by mfussenegger**
## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

**Comment by mfussenegger** [comment]
retest this please

git checkout hiccup

File: server/src/main/java/io/crate/protocols/ssl/SslSettings.java [hunk 1]
```diff
@@ -69,8 +69,11 @@ private SslSettings() {
 
         static SSLMode parse(String value) {
             return switch (value.toLowerCase(Locale.ENGLISH)) {
+                // allow true/false as well because YAML `on` is interpreted as true
                 case "on" -> ON;
+                case "true" -> ON;
                 case "off" -> OFF;
+                case "false" -> OFF;
```
Comment (seut) [inline server/src/main/java/io/crate/protocols/ssl/SslSettings.java:76]: shouldn't this parse method be also used at https://github.com/crate/crate/pull/11350/files#diff-1a801fd9101cdf86ecf02380887786c93c57bde1c650e03c5842d02ce20b4f34R363?

File: server/src/test/java/io/crate/auth/AccessControlMayExecuteTest.java [hunk 2]
```diff
@@ -26,6 +26,7 @@
 import io.crate.action.sql.SessionContext;
 import io.crate.analyze.ParamTypeHints;
 import io.crate.user.Privilege;
+import io.crate.user.StubUserManager;
```
Comment (seut) [inline server/src/test/java/io/crate/auth/AccessControlMayExecuteTest.java:29]: unused?

**Review by seut — APPROVED** [review]
👍 added some more minor comments

File: server/src/main/java/io/crate/protocols/ssl/SslSettings.java (hunk 1 above)
Comment (mfussenegger) [reply server/src/main/java/io/crate/protocols/ssl/SslSettings.java:76]: Somehow the link doesn't point me to a specific place?

Comment (seut) [reply server/src/main/java/io/crate/protocols/ssl/SslSettings.java:76]: ah sorry, should link to `Netty4Transport.initChannel(...)`

Comment (mfussenegger) [reply server/src/main/java/io/crate/protocols/ssl/SslSettings.java:76]: This?

https://github.com/crate/crate/blob/a039e1ff335fdc437da85be593280613eee9a3f3/server/src/main/java/org/elasticsearch/transport/netty4/Netty4Transport.java#L313-L314

It's already returning the parsed enum value

Comment (seut) [reply server/src/main/java/io/crate/protocols/ssl/SslSettings.java:76]: ah ok, seems like I understood the usage or `parse` wrongly.

**Pull request merged**